### DISCUSSION TITLE
Disabling metadata queries - original #23203

### DISCUSSION THREAD

**Discussion by rlroyarsa** [dbeaver/dbeaver#35905] [Q&A] [answered]
### Description

Has this actually been fixed (item #23203)? That is, I'm trying to use Version 24.2.1.202409221725 - with various version of the Snowflake JDBC driver, and I can't find any way to disable the extra metadata queries. I've tried all above (unchecking ALL of Metadata: Show row count for tables
Editors: Force object refresh on editor open
Data Editor: Read table metadata (unique keys)
Data Editor: Read table references (foreign keys)
SQL Editor: Save/restore active schema
Code editor: Read database table/columns for semantic analysis
SQL Processing: Refresh active schema after SQL Execution)...

And I still get the extra queries, such as:

show /* JDBC:DatabaseMetaData.getForeignKeys() */ imported keys in database "ZZZZZ";
...

Which causes the whole tool to just hang forever in our environment because of the catalog size here.

Is there a way to truly turn off all the extra meta-data querying?  Was this fix undone by new versions?

Thanks-
R

### DBeaver Version

Community Edition Version 24.2.1.202409221725

### Operating System

Windows 10 Enterprise

### Database and driver

Snowflake snowflake-jdbc-3.14.0.jar

### Steps to reproduce

Run a query - review the Snowflake Query Activity and see all the metadata queries still executing despite unchecking everything that should prevent them.

**Comment by E1izabeth** [comment, marked as answer]
I suggest checking the setting `Do not read tables metadata in SQL and data editors` in `Connection configuration->Metadata`. Does it help?
![image](https://github.com/user-attachments/assets/c091af11-9271-4498-a522-ae45046441a1)

Have you noticed what you are doing when DBeaver sends this query? You can view all queries DBeaver sends to the database in Query Manager. To use it:
1. Open Query Manager
![image](https://github.com/user-attachments/assets/f505f676-a578-4af2-a786-ba6c1273c17f)

2. Turn on showing all types of queries 
![image](https://github.com/user-attachments/assets/8065c206-dad8-4457-b597-daeb1c2f0616)

**Comment by rlroyarsa** [comment]
Thank you !  That last checkbox made it work.
